src/dal_queryset_sequence/fields.py

```diff
@@ -142,24 +140,27 @@ class GenericForeignKeyModelField(QuerySetSequenceModelField):
     """
     Field that generate automatically the view for the QuerySetSequenceSelect2 widget
     """
-    def __init__(self, *args, model_choice=None, **kwargs):
+    def __init__(self, *args, model_choice=None, widget=None, view=None, **kwargs):
         if model_choice:
             self.model_choice = model_choice
             models_queryset = [model[0].objects.all() for model in model_choice]
             kwargs['queryset'] = QuerySetSequence(*models_queryset)
 
+        if isinstance(widget, type) and isinstance(view, type):  # check if they are classes
+            self.widget_obj = widget
+            self.view_obj = view
+        else:
+            raise AttributeError("Class object are required (not instantiated)")
+
         super().__init__(*args, **kwargs)
 
     def as_url(self, form):
```

Comment:
Tried to handle custom widgets and views, but only if the wdget takes an url as argument and if the view can take a queryset argument. Maybe this is not a good idea, and i could just delete this class and just have  `Select2GenericForeignKeyModelField`

Reply:
Ok cool then if you thinks its ok, `Select2GenericForeignKeyModelField` could inherit from `GenericForeignKeyModelField` and i have to update the docs in consequence. 
So you are ok with the fact that `GenericForeignKeyModelField` may not handle all kind of widgets ?

Reply:
Of course, a form field is not always compatible with the widget class its passed, what would happen if you use a multiple select widget on a single choice field ? Nothing good. So for me it's ok.

Reply:
Ok, then i polish it and it will be ready to be merged ;)